packages/angular_devkit/build_angular/src/angular-cli-files/plugins/index-html-webpack-plugin.ts

```diff
@@ -67,41 +68,38 @@ export class IndexHtmlWebpackPlugin {
 
       const loadOutputFile = (name: string) => compilation.assets[name].source();
 
-      // Get all files for selected entrypoints
-      const unfilteredSortedFiles: string[] = [];
+      // Get all files for script references
+      const files = getEmittedFiles(compilation);
+      const compiledFiles = files.map(f => ({
+        file: f.file,
+        type: 'none',
+        entry: f.name || '',
+      } as CompiledFileInfo));
+
+
       const noModuleFiles = new Set<string>();
       const otherFiles = new Set<string>();
-      for (const entryName of this._options.entrypoints) {
-        const entrypoint = compilation.entrypoints.get(entryName);
-        if (entrypoint && entrypoint.getFiles) {
-          const files: string[] = entrypoint.getFiles() || [];
-          unfilteredSortedFiles.push(...files);
-
-          if (this._options.noModuleEntrypoints.includes(entryName)) {
-            files.forEach(file => noModuleFiles.add(file));
-          } else {
-            files.forEach(file => otherFiles.add(file));
-          }
+
+      // Find all files which have been created out of configured noModule entry points
+      compiledFiles.forEach(f => {
```

Comment:
I strongly suggest that we don't do any index modifications as part of this PR. We should keep that to another PR.

The logic here is different then the previous one as `compiledFiles` will return all files that have been emitted by webpack and not just the entrypoints code.

This includes, images, lazy loaded js, lazy loaded routes, lazy loaded css. which in this case need to be filtered out and not written by the index generator.

We need tests on for this, as this currently breaks the current workflow and hence, we cannot even merge this PR if we turn off the feature.

How are you filling out lazy loaded js files?